File: quiz_app_flutter/lib/presentation/result/model/result_screen_data.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:quiz_app_flutter/domain/entities/result/result_data.dart';

part 'result_screen_data.freezed.dart';

part 'result_screen_data.g.dart';

@freezed
class ResultScreenData with _$ResultScreenData {
  const factory ResultScreenData({
    String? quizTitle,
    String? quizDescription,
    required List<ResultItem> correctItems,
    required List<ResultItem> skippedItems,
    required List<ResultItem> incorrectItems,
    required int totalCorrectItems,
    required int totalSkippedItems,
    required int totalInCorrectItems,
    required int totalQuestions,
    required int resultPercentage,
  }) = _ResultScreenData;

  factory ResultScreenData.fromJson(Map<String, dynamic> json) => _$ResultScreenDataFromJson(json);
}

File: quiz_app_kmp/composeApp/src/commonMain/kotlin/com/tnm/quizappkmp/presentation/result/model/ResultScreenData.kt
package com.tnm.quizappkmp.presentation.result.model

import com.tnm.quizappkmp.domain.model.result.ResultData
import kotlinx.serialization.Serializable

@Serializable
data class ResultScreenData(
    val quizTitle: String?,
    val quizDescription: String? = null,

    val correctItems: List<ResultData.Item>,
    val skippedItems: List<ResultData.Item>,
    val incorrectItems: List<ResultData.Item>,

    val totalCorrectItems: Int,
    val totalSkippedItems: Int,
    val totalInCorrectItems: Int,

    val totalQuestions: Int,
    val resultPercentage: Int,
)